File: src/character.ts
export interface Character {
    name: string,
    title: string,
    maxLevel: number,
    maxSALevel: number,
    rarity: Rarities,
    class: Classes,
    type: Types,
    cost: number,
    id: string,
    imageURL: string,
    leaderSkill: string,
    ezaLeaderSkill?: string,
    superAttack: string,
    ezaSuperAttack?: string,
    ultraSuperAttack?: string,
    ezaUltraSuperAttack?: string,
    passive: string,
    ezaPassive?: string,
    activeSkill?: string,
    activeSkillCondition?: string,
    ezaActiveSkill?: string,
    ezaActiveSkillCondition?: string,
    transformationCondition?: string,
    links: string[],
    categories: string[],
    kiMeter: string[],
    baseHP: number,
    maxLevelHP: number,
    freeDupeHP: number,
    rainbowHP: number,
    baseAttack: number,
    maxLevelAttack: number,
    freeDupeAttack: number,
    rainbowAttack: number,
    baseDefence: number,
    maxDefence: number,
    freeDupeDefence: number,
    rainbowDefence: number,
    kiMultiplier: string,
    transformations?: Transformation[]
}

export enum Classes {
    Super = "Super",
    Extreme = "Extreme"
}

export enum Types {
    PHY = "PHY",
    STR = "STR",
    AGL = "AGL",
    TEQ = "TEQ",
    INT = "INT"
}

export enum Rarities {
    UR = "UR",
    LR = "LR"
}

export interface Transformation {
    transformedID: string,
    transformedName: string,
    transformedClass: Classes,
    transformedType: Types,
    transformedSuperAttack: string,
    transformedEZASuperAttack?: string,
    transformedUltraSuperAttack?: string,
    transformedEZAUltraSuperAttack?: string,
    transformedPassive: string,
    transformedEZAPassive?: string,
    transformedActiveSkill?: string,
    transformedActiveSkillCondition?: string,
    transformedLinks:string[],
    transformedImageURL: string,
}

File: restapi/src/api/character.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Deserialize, Serialize)]
struct Character {
    name: String,
    title: String,
    maxLevel: number,
    maxSALevel: number,
    rarity: Rarities,
    class: Classes,
    type: Types,
    cost: number,
    id: String,
    imageURL: String,
    leaderSkill: String,
    ezaLeaderSkill?: String,
    superAttack: String,
    ezaSuperAttack?: String,
    ultraSuperAttack?: String,
    ezaUltraSuperAttack?: String,
    passive: String,
    ezaPassive?: String,
    activeSkill?: String,
    activeSkillCondition?: String,
    ezaActiveSkill?: String,
    ezaActiveSkillCondition?: String,
    transformationCondition?: String,
    links: String[],
    categories: String[],
    kiMeter: String[],
    baseHP: number,
    maxLevelHP: number,
    freeDupeHP: number,
    rainbowHP: number,
    baseAttack: number,
    maxLevelAttack: number,
    freeDupeAttack: number,
    rainbowAttack: number,
    baseDefence: number,
    maxDefence: number,
    freeDupeDefence: number,
    rainbowDefence: number,
    kiMultiplier: String,
    transformations: Option<Array<Transformation>>
}

 enum Classes {
    Super = "Super",
    Extreme = "Extreme"
}

 enum Types {
    PHY = "PHY",
    STR = "STR",
    AGL = "AGL",
    TEQ = "TEQ",
    INT = "INT"
}
 enum Rarities {
    UR = "UR",
    LR = "LR"
}

struct Transformation {
    transformedID: String,
    transformedName: String,
    transformedClass: Classes,
    transformedType: Types,
    transformedSuperAttack: String,
    transformedEZASuperAttack: Option<String>,
    transformedUltraSuperAttack: Option<String>,
    transformedEZAUltraSuperAttack: Option<String>,
    transformedPassive: String,
    transformedEZAPassive: Option<String>,
    transformedActiveSkill: Option<String>,
    transformedActiveSkillCondition: Option<String>,
    transformedLinks: Array<String>,
    transformedImageURL: String,
}